### PR TITLE
[typescript-angular] remove outdated angular samples

### PR DESCRIPTION
**Pull request by mkusaka**
According to [angular's support policy](https://angular.io/guide/releases#support-policy-and-schedule), versions 9 and 10 are already out of support, and 11 will finish LTS this May.

Therefore, I think we see no need to maintain any more samples of these versions.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02) @davidgamero (2022/03)

### CODE REVIEW

**Comment by akehir** [comment]
I think it's good to remove outdated samples, but I think we're deleting some examples which would still be valid for newer angular versions - and I'd probably suggest to keep those around.

Referring tho these configurations:
- oneOf
- providedIn root / any
- providedInRoot with npm.

Instead of making 1 sample per ng version, we should probably just have 1 generic example each.

**Comment by akehir** [comment]
Since peovidedIn root should be the default, we probably don't nees the following samples, contrary to my previous comment:

> * providedIn root
> * providedInRoot with npm.

But I think oneOf and peovidedIn any might still be useful.

**Comment by mkusaka** [comment]
@akehir Thank you for your review :)

> we should probably just have 1 generic example each.

Sorry. I wasn't sure of a specific way to accomplish this, so I added oneOf and provided in any examples  for ng 12.x and 13.x at 011a8e9.

File: pom.xml [hunk 1]
```diff
@@ -1206,7 +1206,6 @@
                 <module>samples/client/petstore/typescript-nestjs-v6-provided-in-root</module>-->
                 <!-- comment out due to error `npm run build`
                 <module>samples/client/petstore/typescript-jquery/npm</module>-->
-                <module>samples/client/petstore/typescript-angular-v11-provided-in-root</module>
```
Comment (macjohnny) [inline pom.xml:1209]: Please copy these unit tests to the newest amgular version, otherwise we dont have any angular typescript unit tests

**Review by macjohnny — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (mkusaka) [reply pom.xml:1209]: @macjohnny Thanks for your review.
OK. I've just done that at 79484ce (Migration was done with reference to https://update.angular.io/)

**Comment by mkusaka** [comment]
Oh.  Now, https://github.com/OpenAPITools/openapi-generator/pull/11471 added test for angular v13 👍

I'm going to remove the v13 test that was implemented in this PR.

**Comment by wing328** [comment]
@mkusaka can you please resolve the merge conflicts when you've time?

**Comment by mkusaka** [comment]
@wing328  Thanks for remind me.
I've just resolved the conflict and updated the samples 😄

**Pull request merged**